integration-tests/maven/src/test/java/io/quarkus/maven/it/JarRunnerIT.java

```diff
@@ -28,7 +28,9 @@ public void testThatJarRunnerConsoleOutputWorksCorrectly() throws MavenInvocatio
         File testDir = initProject("projects/classic", "projects/project-classic-console-output");
         RunningInvoker running = new RunningInvoker(testDir, false);
 
-        MavenProcessInvocationResult result = running.execute(Arrays.asList("package", "-DskipTests"), Collections.emptyMap());
+        MavenProcessInvocationResult result = running.execute(
+                Arrays.asList("package", "-DskipTests", "-Dquarkus.package.main-class=foo.bar.MainClass"),
+                Collections.emptyMap());
```

Comment:
Please add another test when you do that.

Reply:
I think the test in here is enough as without this patch if you add the mainClass property (via build tool e.g maven or in application.properties )you could not launch the application - https://github.com/quarkusio/quarkus/blob/95590681ea0a9c94be8f81a16114e0eca0495ab5/integration-tests/maven/src/test/java/io/quarkus/maven/it/JarRunnerIT.java#L44-L47.

Reply:
I think this should really be a different test to avoid confusion. I also don't understand how this test passes? Where does foo.bar.MainClass come from?

Reply:
Yes, we can have another test. I wanted to quickly add the mainClass property when packaging in this existing JarRunner test to avoid straining the CI with yet another similar test. 

foo.bar.MainClass is just an arbitrary class name supplied as configuration property to override the default ‘io.quarkus.runner.GeneratedMain’

Reply:
But the class doesn't exist so why does the test pass?

Reply:
It passes because of my second commit https://github.com/quarkusio/quarkus/pull/5542/commits/a069ecba35fcd5d9cf0f977b3ec9d091b2233562

Reply:
Is this how the mainClass property should be used? Or it is a user defined one?